app/lib/actions/pulp3/capsule_content/generate_metadata.rb

```diff
@@ -6,6 +6,7 @@ class GenerateMetadata < Pulp3::AbstractAsyncTask
         def plan(repository, smart_proxy, options = {})
           options[:contents_changed] = (options && options.key?(:contents_changed)) ? options[:contents_changed] : true
           sequence do
+            plan_action(RefreshContentGuard, smart_proxy) unless repository.unprotected
```

Comment:
i think it would be better to put this at more of a global  capsule sync level, that way it only executes once per smart_proxy.

Reply:
even though it may run regardless of whether its needed (without the 'unless repository.unprotected') it'd only run once